File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        int prev = 0;
        StringBuilder str = new StringBuilder();
        for(int i : spaces)
        {
            str.append(s.substring(prev,i));
            str.append(" ");
            prev = i;
        }
        str.append(s.substring(prev));
        return str.toString();
    }
}

File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.cpp
class Solution {
    public :
    string addSpaces(string s, vector<int>& spaces) {
        int prev = 0;
        string str = "";
        for(int i : spaces)
        {
            str += s.substr(prev,i-prev);
            str += " ";
            prev = i;
        }
        str += s.substr(prev);
        return str;
    }
};